### PR TITLE
chore: use stricter typescript rules on the main package

### PR DESCRIPTION
**Pull request by benoitf**
### What does this PR do?
Enable `"@tsconfig/strictest"` typescript rule on the main package to enforce more checks int he source code

but disable for now `exactOptionalPropertyTypes` from this rule

### How to test this PR?

- [x] Tests are covering the bug fix or the new feature

### CODE REVIEW

File: packages/main/src/plugin/container-registry.ts [hunk 1]
```diff
@@ -1070,7 +1072,7 @@ export class ContainerProviderRegistry {
 
   getImageName(inspectInfo: Dockerode.ImageInspectInfo): string {
     const tags = inspectInfo.RepoTags;
-    if (!tags) {
+    if (!tags?.[0]) {
```
Comment (axel7083) [inline packages/main/src/plugin/container-registry.ts:1075]:
```suggestion
    if (!tags || tags.length === 0 || !tags[0]) {
```

**Review by axel7083 — APPROVED** [review]
Thanks for the works done :rocket: kudos for taking the time to do it.

> I can't really read and check every change due to the size of the PR, but changes LGTM

File: packages/main/src/plugin/container-registry.ts (hunk 1 above)
Comment (benoitf) [reply packages/main/src/plugin/container-registry.ts:1075]: hi @axel7083 , could you give me why we would need these 3 condition rather than one ?

Comment (benoitf) [reply packages/main/src/plugin/container-registry.ts:1075]: I think when editing at some point I received an error telling me to merge the condition as well

Comment (axel7083) [reply packages/main/src/plugin/container-registry.ts:1075]: image

I just checked, and there is no need, as javascript do not raise error while accessing index that does not exists !

Comment (benoitf) [reply packages/main/src/plugin/container-registry.ts:1075]: yes the optional chaining is checking that for us

File: packages/main/src/plugin/kubernetes-client-kubeconfig.spec.ts [hunk 2]
```diff
@@ -109,7 +113,16 @@ describe('context tests', () => {
...
     client.saveKubeConfig = vi.fn().mockImplementation((_config: KubeConfig) => {});
 
+    if (!originalContexts[0]?.name) {
+      throw new Error('originalContexts[0].name should be defined');
+    }
+
     await client.deleteContext(originalContexts[0].name);
+
+    if (!originalContexts[1]?.name) {
+      throw new Error('originalContexts[0].name should be defined');
```
Comment (lstocchi) [inline packages/main/src/plugin/kubernetes-client-kubeconfig.spec.ts]:
```suggestion
      throw new Error('originalContexts[1].name should be defined');
```

**Review by lstocchi — APPROVED** [review]
LGTM

**Pull request merged**